oemof/core/network/entities/__init__.py

```diff
@@ -46,6 +46,26 @@ def __init__(self, **kwargs):
         self.excess_costs = kwargs.get('excess_costs', 0)
         self.shortage_costs = kwargs.get('shortage_costs', 10e10)
 
+        if self.excess == True:
+           self.create_excess_slack_component()
+        if self.shortage == True:
+           self.create_shortage_slack_component()
+
+    def create_excess_slack_component(self):
+        """
+        """
+        excess = ExcessSlack(uid=self.uid+'_excess',
+                             inputs=[self],
+                             costs=self.excess_costs)
+        return excess
+
+    def create_shortage_slack_component(self):
+        """
+        """
+        shortage = ShortageSlack(uid=self.uid+'_shortage',
+                                 outputs=[self],
+                                 costs=self.shortage_costs)
+        return shortage
```

Comment:
In the long run I would remove this part because it makes it less clear to create a component together with the bus. You could also define a sink object (demand) in this way. The bus will be more complicate and the structure less clear.

I would add a warning so that we can remove it in the long run.